src/core/exec-invoke.c

```diff
@@ -4302,13 +4302,18 @@ int exec_invoke(
                 return log_exec_error_errno(context, params, r, "Failed to set up standard input: %m");
         }
 
-        r = setup_output(context, params, STDOUT_FILENO, socket_fd, named_iofds, basename(command->path), uid, gid, &journal_stream_dev, &journal_stream_ino);
+        _cleanup_free_ char *fname = NULL;
+        r = path_extract_filename(command->path, &fname);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to extract filename from path '%s', ignoring: %m", command->path);
```

Comment:
you are propagating the error here, i.e. fail the invocation. hence it shouldn#t say "ignoring".

And it should be at error log level, not warning.

And please use log_exec_error_errno() like in the other cases

and please set `*exit_status = EXIT_STDOUT` or so?

Reply:
Thanks for taking the time to look at it, I'll make sure to get to it. And just to be clear because i wasn't entirely sure, I would follow the conventions of the other logs, right?

so something like
`log_exec_error_errno(context, params, r, ..., command->path)`

or would this suffice?

`log_exec_error_errno(r, ..., command->path)`